audbackend/core/backend.py

```diff
@@ -321,28 +321,71 @@ def latest_version(
 
     def _ls(
             self,
-            path: str,
-    ) -> typing.List:  # pragma: no cover
-        r"""List content of path."""
+            folder: str,
+    ) -> typing.List[typing.Tuple[str, str, str]]:  # pragma: no cover
+        r"""List all files under folder."""
         raise NotImplementedError()
 
     def ls(
             self,
-            path: str,
-    ) -> typing.List[str]:
-        r"""List content of path.
+            folder: str = '/',
+            *,
+            latest_version: bool = False,
+    ) -> typing.List[typing.Tuple[str, str, str]]:
+        r"""List all files under folder.
+
+        Returns a sorted list of tuples
+        with path, extension and version.
+        If file extension is not empty,
+        it starts with a dot.
 
         Args:
-            path: relative path to folder in repository
+            folder: folder on backend
+            latest_version: if multiple versions of a file exist,
+                only include the latest
 
         Returns:
-            folder content
+            list of tuples (path, extension, version)
 
         Raises:
-            RuntimeError: if ``path`` does not exist on backend
+            FileNotFoundError: if ``folder`` does not exist
+            ValueError: if ``folder`` contains invalid character
 
-        """
-        return sorted(self._ls(path))
+        Example:
+            >>> backend.ls('folder')[:2]
+            [('folder/name.ext', '.ext', '1.0.0'), ('folder/name.ext', '.ext', '2.0.0')]
+            >>> backend.ls('folder', latest_version=True)[:1]
+            [('folder/name.ext', '.ext', '2.0.0')]
+
+        """  # noqa: E501
+        utils.check_path_for_allowed_chars(folder)
+        if not folder.endswith('/'):
+            folder += '/'
+        paths = self._ls(folder)
+        paths = sorted(paths)
+
+        if len(paths) == 0:
+            if folder == '/':  # pragma: no cover
+                # special case that there are no files on the backend
```

Comment:
Sounds like we should be able to create a test for it, e.g. creating a new empty repo?

Reply:
Instead of rewirting the tests, you could add an additional test that creates a new empty repo just for this particular test?

Reply:
I anyway wanted to rewrite the tests to ensure the backend is cleaned after each test. This is now automatically done by the `backend` fixture. I now added a test for calling `ls()` on an empty repository and removed the pragma comment.

Reply:
I also simplified the tests in a couple of places by using `audeer.touch()` and not encoding the unique ID in the path but already in the repository.

Reply:
And we now keep the list of backends that will be tested in `conftest.py`. So during local debugging the developer can comment out certain backends to speed up the tests.